src/registrar/management/commands/populate_domain_updated_federal_agency.py

```diff
@@ -0,0 +1,120 @@
+        "Japan-US Friendship Commission": "Japan-U.S. Friendship Commission",
+        "Japan-United States Friendship Commission": "Japan-U.S. Friendship Commission",
+        "John F. Kennedy Center for Performing Arts": "John F. Kennedy Center for the Performing Arts",
+        "Occupational Safety & Health Review Commission": "Occupational Safety and Health Review Commission",
+        "Corporation for National & Community Service": "Corporation for National and Community Service",
+        "Export/Import Bank of the U.S.": "Export-Import Bank of the United States",
+        "Medical Payment Advisory Commission": "Medicare Payment Advisory Commission",
+        "U.S. Peace Corps": "Peace Corps",
+        "Chemical Safety Board": "U.S. Chemical Safety Board",
+        "Nuclear Waste Technical Review Board": "U.S. Nuclear Waste Technical Review Board",
+        "State, Local, and Tribal Government": "Non-Federal Agency",
+    }
+
+    def handle(self, **options):
+        """
+        Renames the Federal Agency to the correct new naming
+        for both Domain Information and Domain Requests objects.
+
+        NOTE: If it's NULL for a domain request, we skip it as
+        a user most likely hasn't gotten to it yet.
+        """
+        logger.info("Transferring federal agencies to FederalAgency object")
+        # DomainInformation object we populate with updated_federal_agency which are then bulk updated
+        domain_infos_to_update = []
+        domain_requests_to_update = []
+        # Domain Requests with null federal_agency that are not populated with updated_federal_agency
+        domain_requests_skipped = []
+        domain_infos_with_errors = []
+        domain_requests_with_errors = []
+
+        domain_infos = DomainInformation.objects.all()
+        domain_requests = DomainRequest.objects.all()
+
+        logger.info(f"Found {len(domain_infos)} DomainInfo objects with federal agency.")
+        logger.info(f"Found {len(domain_requests)} Domain Request objects with federal agency.")
+
+        for domain_info in domain_infos:
+            try:
+                federal_agency_row = self.find_federal_agency_row(domain_info)
+                domain_info.updated_federal_agency = federal_agency_row
+                domain_infos_to_update.append(domain_info)
+                logger.info(
+                    f"DomainInformation {domain_info} updated_federal_agency set to:"
+                    f"{domain_info.updated_federal_agency}"
+                )
+            except Exception as err:
+                logger.info(f"DomainInformation for {domain_info} failed to update updated_federal_agency: {err}")
+                domain_infos_with_errors.append(domain_info)
+
+        ScriptDataHelper.bulk_update_fields(DomainInformation, domain_infos_to_update, ["updated_federal_agency"])
+
+        for domain_request in domain_requests:
+            try:
+                if not domain_request.federal_agency:
```

Comment:
Nope! 

The for loop for `domain_information` does not need this check as it's handled in `find_federal_agency_row` (defaults updated_federal_agency to "Non-Federal Agency") and Domain Requests is a special case we handle here as since a null Federal Agency likely implies that the requestor hasn't gotten to that portion of the Domain Request yet (hence why we skip updating Domain Requests with null Federal Agency) yet update Domain Information to "Non-Federal Agency"

Reply:
**blocking** @therealslimhsiehdyand @erinysong shouldn't you actually check if federal_agency is actually None. This will skip ones that are just false, and for false ones you want updated_federal_agency to really be the Non-Federal agency foreign key. That seems to be missing here.

Reply:
Resolved in [this commit](00b99d8786e5f54e512f8fed12843aa4f9680612) -- great catch!